File: sportseq-web/src/api/equipment/warehouse.ts
import { del, get, post, put } from '../request'

export interface WarehouseItem {
  id: number
  warehouseCode: string
  warehouseName: string
  manager: number | null
  phone: string | null
  address: string | null
  status: number
  remark: string | null
}

export interface WarehouseDTO {
  id?: number
  warehouseCode: string
  warehouseName: string
  manager?: number
  phone?: string
  address?: string
  status?: number
  remark?: string
}

export const getWarehouseList = () => get<WarehouseItem[]>('/equipment/warehouse/list')

export const addWarehouse = (data: WarehouseDTO) => post<void>('/equipment/warehouse', data)

export const updateWarehouse = (data: WarehouseDTO) => put<void>('/equipment/warehouse', data)

export const deleteWarehouse = (id: number) => del<void>(`/equipment/warehouse/${id}`)

File: sportseq-admin/src/main/java/com/company/sportseq/entity/Warehouse.java
package com.company.sportseq.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.company.sportseq.common.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("warehouse")
public class Warehouse extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String warehouseCode;
    private String warehouseName;
    private Long manager;
    private String phone;
    private String address;
    private Integer status;
    private String remark;
}
